tiledb/sm/array_schema/array_schema.cc

```diff
@@ -964,8 +964,10 @@ ArraySchema ArraySchema::deserialize(
   // #TODO Add security validation
   auto version = deserializer.read<uint32_t>();
   if (!(version <= constants::format_version)) {
-    throw ArraySchemaStatusException(
-        "Failed to deserialize array schema; Incompatible format version.");
+    std::stringstream ss;
+    ss << "Failed to deserialize array schema; Incompatible format version: ";
+    ss << "got " << version << "; expected <= " << constants::format_version;
```

Comment:
It may be more useful to users to word as:
```
ss << "Failed to deserialize array schema; Format version ";
ss << version << " is incompatible with expected format version ";
ss << constants::format_version;
```

Reply:
ok thanks! i do want `<=` rather than just "incompatible" ...